app/src/main/java/app/musikus/ui/activesession/ActiveSessionUiState.kt

```diff
@@ -8,20 +8,144 @@
 
 package app.musikus.ui.activesession
 
-import app.musikus.database.Nullable
-import app.musikus.database.daos.LibraryFolder
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.graphics.Color
+import app.musikus.database.LibraryFolderWithItems
+import app.musikus.database.UUIDConverter
 import app.musikus.database.daos.LibraryItem
-import app.musikus.ui.library.DialogMode
-import app.musikus.ui.library.LibraryItemDialogUiEvent
-import app.musikus.ui.library.LibraryItemDialogUiState
-import app.musikus.ui.library.LibraryItemEditData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import java.time.ZonedDateTime
 import java.util.UUID
-import kotlin.time.Duration
 
 
+// TODO UIText for all Strings?
 
+@Stable
+enum class SessionPausedResumedState {
+    RUNNING,
+    PAUSED
+}
+
+/** Specific one-time events triggering some UI action in ActiveSession */
+@Stable
+data class ActiveSessionEventStates (
+    val sessionSaved: Boolean = false,
```

Comment:
We should also talk about whether we want default values for UiStates or not and then make it consistent across the app.

Reply:
I think default values are quite useful. Mainly, I set them all while doing the previews to not having to pass all parameters to every Composable.

Having default values is also a common practice amongst many guides I found (e.g. [here](https://medium.com/@MrAndroid/android-architecture-with-mvvm-and-uistate-f29aa5494465) or [here](https://craigrussell.io/2017/11/30/representing-view-state-with-kotlin-data-classes)) and even [Google does it](https://github.com/google-developer-training/basic-android-kotlin-compose-training-unscramble/blob/viewmodel/app/src/main/java/com/example/unscramble/ui/GameUiState.kt)

I see the following problems when not having default values:
* Everytime a StateFlow is created, in the `stateIn`, all parameters must be passed as default. This will result in an enourmous augmentation of boilerplate code which makes it even more convoluted than it already is (because of `stateIn` boilerplate) without adding any functionality. This will make developing experience much worse IMO
* For previews, it's really cumbersome having to pass all parameters to every UI state and slows down creation of previews

Plus it comes with the benefit of having all default values easily visible in one place along the class definition itself.

Reply:
Yes, the boilerplate code is annoying. However, the "assembly" of the UiState doesn't/shouldn't contain much or any logic and thus does not need to be maintained so much. The `combineStateFlow` wrapper seems like a good idea.

Not using default values does have some advantages in my mind:
- Developers are forced to assign values to each property inside the `combine` function.
- Most UiStates are instantiated only once (twice if you count the `stateIn` separately). Because of that default all the values for a UiState are in one place, while the class declaration focuses solely on the **structure** of the UiState.
- The default values used for the UiState are mostly empty strings and lists which are not useful for previews anyways, right?

That's all I can think of for the moment. Most importantly I would like to keep it consistent within the app so maybe we change it for this PR?

Reply:
The big issue was the Previews. E.g. if you want to create a preview of the main screen you would also have to provide dummy data for every dialog ui state or secondary ui state you don't show in your preview. But since we may change to nullable States for non-permanently visible elements anyways we can think about it.

Also, the stateIn default arg clutter was also a big reason. Maybe we can circumvent it with the combine wrapper (although it does not appear to be universally adaptable to any number of input flows) or we can define the default UI states seperately and just pass the default variable (the default of `stateIn` actually is in 99,9% irrellevant anyways except there is an animation or so which delays distributing new values...)

Reply:
> Am I wrong or isn't the "assembly" where most/all of the ui logic is implemented?
You are not wrong, assembling the UiState does require some logic. What I meant is that any major data processing logic should happen inside a _use case_ and what you call 'ui logic' is more a kind of 'extraction' of Integers and Strings from the data provided by use cases, _ViewModelState_ and combinations thereof.

> The big issue was the Previews. E.g. if you want to create a preview of the main screen you would also have to provide dummy data for every dialog ui state or secondary ui state you don't show in your preview. But since we may change to nullable States for non-permanently visible elements anyways we can think about it.
Ok, now it makes sense why you would definitely need  default  values. However, if we set the convention, that all UiStates that describe a UI element which is not shown are null, then we don't have that problem anymore, do we?

> Also, the stateIn default arg clutter was also a big reason. Maybe we can circumvent it with the combine wrapper (although it does not appear to be universally adaptable to any number of input flows) or we can define the default UI states seperately and just pass the default variable (the default of `stateIn` actually is in 99,9% irrellevant anyways except there is an animation or so which delays distributing new values...)
I totally agree it's annoying... Maybe we can just collapse the `stateIn` in Android Studio?